coveo-settings/tests_settings/test_settings.py

```diff
@@ -30,6 +32,16 @@ def _clean_environment_variable(*environment_variable_name: str) -> None:
             del os.environ[variable_name]
 
 
+@contextmanager
+def _clean_environment_context(*environment_variable_name: str) -> Generator[None, None, None]:
+    """ Context manager for the above function. """
+    try:
+        _clean_environment_variable(*environment_variable_name)
+        yield
+    finally:
+        _clean_environment_variable(*environment_variable_name)
```

Comment:
The behavior surprised me. At first I thought you would clean the environment, `yield`, and then restore the environment to what is was before this context was called. However, you call `_clean_environment_variable(*environment_variable_name)` twice. I checked how you use it below, and I think that this is your intent. Hence (if it really is your intent), the function should obviously remain as it is, but I would state it in the doc:

```suggestion
    """ Removes one or many environment variables before and after running the block. """
    try:
        _clean_environment_variable(*environment_variable_name)
        yield
    finally:
        _clean_environment_variable(*environment_variable_name)
```